hack/generate.sh

```diff
@@ -32,6 +32,6 @@ else
         --volume "${PWD}:${WORKDIR}:rw,z" \
         --entrypoint sh \
         --workdir "${WORKDIR}" \
-        docker.io/golang:1.22 \
+        quay.io/metal3-io/basic-checks:golang-1.22 \
```

Comment:
```suggestion
        --pull=always \
        quay.io/metal3-io/basic-checks:golang-1.22 \
```
Otherwise locally we never will get updates made to that image after the first pull is done. For golang image it is not meaningful as its just bug fixes, but here if we add a new package here, it wouldn't show up locally unless manually forcing cleanup/repull.